pkg/odo/cli/component/common_link.go

```diff
@@ -25,7 +25,6 @@ const unlink = "unlink"
 
 type commonLinkOptions struct {
 	wait             bool
-	port             string
```

Comment:
This causes an error during the Make.
```sh
➜  make; make install
go build -mod=vendor -ldflags="-X github.com/openshift/odo/pkg/version.GITCOMMIT=52f9ab91b" cmd/odo/odo.go
# github.com/openshift/odo/pkg/odo/cli/component
pkg/odo/cli/component/unlink.go:101:42: o.port undefined (type *UnlinkOptions has no field or method port)
make: *** [Makefile:57: bin] Error 2
go install -mod=vendor -ldflags="-X github.com/openshift/odo/pkg/version.GITCOMMIT=52f9ab91b" ./cmd/odo/
# github.com/openshift/odo/pkg/odo/cli/component
pkg/odo/cli/component/unlink.go:101:42: o.port undefined (type *UnlinkOptions has no field or method port)
make: *** [Makefile:61: install] Error 2                  
```

Reply:
This PR seems overwhelming. Do you think you should keep the Link cleanup for another PR? For now, you can just remove the references from comments and help docs, do the code cleanup in another PR like you said earlier, WDYT?

Reply:
change reset